packages/typespec-python/src/types.ts

```diff
@@ -112,7 +121,7 @@ export function getType(
         case "Union":
         case "ModelProperty":
         case "UnionVariant":
-            return getType(context, getClientType(context, type));
+            return getType(context, getClientType(context, type), getDecoratorInfo(context.program, type));
```

Comment:
In the future, hope TCGC can add decorator info for `modelProperty`
cc @tadelesh

Reply:
there is a design where we're going to add it onto path parameters, because that's the only validation we're thinking of doing. That's part of the operation and client work on tcgc